### PR TITLE
declare components type

### PR DESCRIPTION
**Pull request by ajuner**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch. Pull request will be merged after one of collaborators approve. Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/vueComponent/ant-design-vue/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### Additional Plan? (Optional if not new feature)

继续补充和添加了ts声明
在检查的时候发现Form组件的handleFinishFailed内scrollToField方法被调用时传入的是数组，但scrollToField方法接收的并不是数组，这里进行了一个小修改。

### CODE REVIEW

File: components/form/Form.tsx [hunk 1]
```diff
@@ -154,8 +154,8 @@ const Form = defineComponent({
     validate(...args: any[]) {
       return this.validateField(...args);
     },
-    scrollToField(name: string | number, options = {}) {
-      const fields = this.getFieldsByNameList([name]);
+    scrollToField(name: InternalNamePath, options = {}) {
```
Comment (tangjinzhou) [inline components/form/Form.tsx:157]: 应该支持 `string | number | (string | number)[]`
NamePath 类型

File: components/menu/index.tsx [hunk 2]
```diff
@@ -29,8 +29,8 @@ export const menuProps = {
   selectable: PropTypes.looseBool,
   selectedKeys: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.string, PropTypes.number])),
   defaultSelectedKeys: PropTypes.array,
-  openKeys: PropTypes.array,
-  defaultOpenKeys: PropTypes.array,
+  openKeys: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.string])),
```
Comment (tangjinzhou) [inline components/menu/index.tsx:32]: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.string, PropTypes.number]))

Comment (tangjinzhou) [reply components/menu/index.tsx:32]: defaultOpenKeys 也一样

File: components/menu/index.tsx [hunk 3]
```diff
@@ -88,7 +88,7 @@ const Menu = defineComponent({
       'Menu',
       "`inlineCollapsed` should only be used when Menu's `mode` is inline.",
     );
-    let sOpenKeys;
+    let sOpenKeys: string[];
```
Comment (tangjinzhou) [inline components/menu/index.tsx:91]: 支持 number

Comment (tangjinzhou) [reply components/menu/index.tsx:91]: 如下都是要支持 number

**Review by tangjinzhou — CHANGES_REQUESTED** [review]

**Comment by ajuner** [comment]
ok

**Comment by ajuner** [comment]
@tangjinzhou 修复啦，多谢指点

**Pull request merged**